argo/kfp-compiler/kfp_compiler/compiler.py

```diff
@@ -70,35 +70,44 @@ def load_fn(tfx_components, env={}):
 
 @click.command()
 @click.option('--pipeline_config', help='Pipeline configuration in yaml format', required=True)
+@click.option('--provider_config', help='Provider configuration in yaml format', required=True)
 @click.option('--output_file', help='Output file path', required=True)
-@click.option('--execution_mode', type=click.Choice(['v1', 'v2']), help='KFP Execution mode', required=True)
-def compile(pipeline_config, output_file, execution_mode):
+def compile(pipeline_config, provider_config, output_file):
     """Compiles TFX components into a Kubeflow Pipelines pipeline definition"""
-    with open(pipeline_config, "r") as stream:
-        config = yaml.safe_load(stream)
+    with open(pipeline_config, "r") as pipeline_stream, open(provider_config, "r") as provider_stream:
+        pipeline_config_contents = yaml.safe_load(pipeline_stream)
+        provider_config_contents = yaml.safe_load(provider_stream)
 
-        click.secho(f'Compiling with config: {config}', fg='green')
+        click.secho(f'Compiling with pipeline: {pipeline_config_contents} and provider {provider_config_contents} ', fg='green')
 
-        components = load_fn(config['tfxComponents'], config.get('env', {}))()
-        expanded_components = expand_components_with_pusher(components, config['servingLocation'])
+        pipeline_root, serving_model_directory, temp_location = pipeline_paths_for_config(pipeline_config_contents, provider_config_contents)
 
-        compile_fn = compile_v1 if execution_mode == 'v1' else compile_v2
+        beam_args_named_values = provider_config_contents.get('defaultBeamArgs', [])
+        beam_args_named_values.extend(pipeline_config_contents.get('beamArgs', []))
+        beam_args = named_values_to_multimap(beam_args_named_values)
 
-        compile_fn(config, output_file).run(
+        components = load_fn(pipeline_config_contents['tfxComponents'], pipeline_config_contents.get('env', []))()
+        expanded_components = expand_components_with_pusher(components, serving_model_directory)
+
+        compile_fn = compile_v1 if provider_config_contents['executionMode'] == 'v1' else compile_v2
+        beam_cli_args = dict_to_cli_args(beam_args)
```

Comment:
shall we move this up to be with the rest of the beam args code (i.e. line 88)

Reply:
done